packages/adblocker/src/engine/bucket/cosmetic.ts

```diff
@@ -431,6 +415,8 @@ export default class CosmeticFilterBucket {
     getExtendedRules?: boolean;
     getRulesFromDOM?: boolean;
     getRulesFromHostname?: boolean;
+
+    isFilterEligible?: (filter: number) => boolean;
```

Comment:
if we would pass a filter object as an argument instead, we would get a better type safety